src/synthetic-monitoring.js

```diff
@@ -0,0 +1,339 @@
+let appInsights = require("applicationinsights");
+const axios = require('axios');
+const sslCertificate = require('get-ssl-certificate')
+const { TableClient, AzureNamedKeyCredential } = require("@azure/data-tables");
+appInsights.setup(process.env.APP_INSIGHT_CONNECTION_STRING).start();
+
+const account = process.env.STORAGE_ACCOUNT_NAME;
+const accountKey = process.env.STORAGE_ACCOUNT_KEY;
+const tableName = process.env.STORAGE_ACCOUNT_TABLE_NAME
+const availabilityPrefix = process.env.AVAILABILITY_PREFIX
+const httpClientTimeout = process.env.HTTP_CLIENT_TIMEOUT
+
+const credential = new AzureNamedKeyCredential(account, accountKey);
+const tableClient = new TableClient(`https://${account}.table.core.windows.net`, tableName, credential);
+
+let client = new appInsights.TelemetryClient(process.env.APP_INSIGHT_CONNECTION_STRING);
+
+const keysForTelemetry = ['success', 'message', 'duration', 'runLocation'];
+const keysForEvent = ['duration', 'targetStatus', 'targetExpirationTimestamp', 'httpStatus', 'targetTlsVersion', 'targetExpireInDays'] ;
+const keysForEventProperties = ['domain', 'checkCert'] ;
+const TLS_VERSION_KEY = "TLS_VERSION"
+const START_TIMESTAMP_KEY = "x-request-timestamp"
+const RESPONSE_TIME_KEY = "RESPONSE_TIME"
+
+//prepare axios interceptors
+axios.interceptors.response.use(function (response) {
+    //adding tls version to response
+    response[TLS_VERSION_KEY] = response.request.res.socket.getProtocol()
+    response[RESPONSE_TIME_KEY] = Date.now() - response.config.headers[START_TIMESTAMP_KEY]
+    return response;
+  }, function (error) {
+    console.error(`resp error interceptor: ${JSON.stringify(error)}`)
+    //nothing to do
+    return Promise.reject(error);
+  });
+
+axios.interceptors.request.use(
+    (config) => {
+      config.headers[START_TIMESTAMP_KEY] = Date.now();
+      return config;
+    },
+    (error) => {
+      console.error(`req error interceptor: ${JSON.stringify(error)}`)
+      return Promise.reject(error);
+    }
+  );
+
+
+async function main() {
+    let tableEntities = tableClient.listEntities();
+    let tests = []
+    for await (const tableConfiguration of tableEntities) {
+
+      //property names remap and parsing
+      let monitoringConfiguration = {...tableConfiguration}
+      console.log(`monitoringConfiguration: ${JSON.stringify(monitoringConfiguration)}`)
+      monitoringConfiguration['appName'] = tableConfiguration.partitionKey
+      monitoringConfiguration['apiName'] = tableConfiguration.rowKey
+      monitoringConfiguration['tags'] = !isNull(monitoringConfiguration['tags']) ? JSON.parse(monitoringConfiguration['tags']) : {}
```

Comment:
Since `JSON.parse` should raise exception, we should handle exception accordingly